### DISCUSSION TITLE
Font change when export PNG

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2235] [Q&A] [answered]
### Introduction

Hi

I've tried here https://resources.jointjs.com/demos/rappid/apps/KitchenSink/index.html then click `Export PNG` I see font inside image is not the same as in the canvas

How to make Export PNG font align with our design on canvas?

Thank you

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You need to embed the font as base64.

An example can be found in the [Collapsible](https://www.jointjs.com/demos/collapse-expand) demo.

```js
paper.openAsPNG({
    useComputedStyles: false,
    stylesheet: `
        @font-face {
            font-family: 'source_sans_prosemibold';
            src: url(${woff2Base64}) format('woff2'), url(${woffBase64}) format('woff');
            font-weight: 600;
            font-style: normal;
        }
    `
});
```

Also see: 
https://groups.google.com/g/jointjs/c/TaFUZE57nvE
https://groups.google.com/g/jointjs/c/UAoxdXief70/m/4f5ltavGAgAJ
https://stackoverflow.com/questions/11438150/using-font-face-with-an-svg-font-embedded-in-the-current-html-page/11438309#11438309

**Reply by kumilingus** [reply]
Here's an actual example of how to load a font and embed it into the exported SVG / PNG: https://github.com/clientIO/joint/discussions/2239

**Reply by pandika-csgi** [reply]
Thanks @kumilingus, I will try later, that Codepen helps me a lot

**Comment by pandika-csgi** [comment]
Hi @kumilingus , I want to ask about related export to PNG, I've added an image to canvas via HTML not add to the graph, is it possible to show the image that I added on export PNG?
because I won't add the image to the graph

Thank youu

**Reply by kumilingus** [reply]
The image must be a descendant of the paper's `svg`. It can be exported:
- if it's an SVG `<image/>`
- if it's an HTML `<img/>` inside a `<foreignObject/>`
